refactor(parser): shorten code

overlookmotel opened the pull request:
Pure refactor. Shorten code by using `AstBuilder` methods which produce the desired type in a single call.

Boshen commented:
### Merge activity

* **Jan 21, 9:19 AM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Jan 21, 9:19 AM EST**: A user added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).

overlookmotel committed:
refactor(parser): shorten code (#8640)

Pure refactor. Shorten code by using `AstBuilder` methods which produce the desired type in a single call.
